text/0000-landmarks.md

```diff
@@ -0,0 +1,43 @@
+# Landmark observations
+Add landmarks support to Cartographer.
+
+## Summary
+[summary]: #summary
+Cartographer will be able to work with a new type of sensor data that contains IDs and locations of landmarks.
+
+## Motivation
+[motivation]: #motivation
+Landmarks translation/orientation can improve SLAM quality. Landmarks rendered on the map can be useful for inspection. 
+
+## Approach
+[approach]: #approach
+Landmarks are sparse features of geometry that we want to map. Unlike dense range data, landmarks are assumed to be sparse in comparison. In order to support using them, current SLAM approach has to be generalized. 
+
+### Sensor
+Each landmark will have an ID and a position relative to the robot with or without orientation. A landmark observation can contain several landmarks at a time.
+
+### Submap
+Submaps for 2D and 3D geometries are occupancy grids that contain accumulated range data.  There can be two approaches to add landmarks to the submaps:
+
+1. Submap holds a dictionary with landmark data keyed by landmark IDs;
+2. Every cell of the grid is modified to hold various features, for example, likelihood of landmarks, ML features. 
+
+The second approach is more generic and makes it possible to introduce landmarks based on the strength of the signal, e.g. Wi-Fi signal, BT beacons. The first approach should work faster, it is also more flexible and non-intrusive.
+  
+### Constraint builder
```

Comment:
In general I think you should be more concrete what changes you are planning to make to `ConstraintBuilder` and `PoseGraph`.